myscript.js

```diff
@@ -14,15 +14,38 @@ function updateButtons(){
 	if(buttons.length == 0){
 		return;
 	}
+
+	// Remove archive, spam and delete button on the email view
+	var deleteImage =  document.getElementsByClassName("ar9 T-I-J3 J-J5-Ji");
+	if (deleteImage.length > 0) {
+		var deleteImageParent = deleteImage[0].parentNode;
+		deleteImageParent.removeChild(deleteImage[0]);
+		deleteImageParent.innerHTML = "Delete";
+	}
+	
+	var spamImage =  document.getElementsByClassName("asl T-I-J3 J-J5-Ji");
+	if (spamImage.length > 0) {
+		var spamImageParent = spamImage[0].parentNode;
+		spamImageParent.removeChild(spamImage[0]);
+		spamImageParent.innerHTML = "Mark Spam";
+	}
+	
+	var archiveImage = document.getElementsByClassName("ar8 T-I-J3 J-J5-Ji");
+	if (archiveImage.length > 0) {
+		var archiveImageParent = archiveImage[0].parentNode;
+		archiveImageParent.removeChild(archiveImage[0]);
+		archiveImageParent.innerHTML = "Archive";
+	}
+	
```

Comment:
These three `if` blocks can be abstracted into one function that takes a classname string and a new label string, i.e `changeIconLabel("ar9 T-I-J3 J-J5-Ji", "Delete")`